scripts/pocket-change.js

```diff
@@ -380,4 +502,72 @@ export default class PocketChange {
       token.actor.sheet.render(true);
     }
   }
+
+  /**
+   * Converts the provided token to a item piles lootable sheet
+   *
+   * @param {object} options
+   * @param {Token5e} options.token - the token to convert
+   * @param {number} options.userOption - (optional) the type of convertion by default is 1
+   * You've got 4 options to choose from:
+   * 0 = No Special Effect, Coin roll and -if enabled- Item Pile Transformation Only
+   * 1 = Light Effect only
+   * 2 = Change Image Only
+   * 3 = Both Image Change and Light effect
+   * @param {string} options.imgPath - (optional) the path to the image by default is the one set on the module setting
+   * @param {Light} options.light - (optional) explicit light effect to use if none is passed a default one is used
+   */
+  async _convertToItemPiles({
+    token,
+    userOption = 1,
+    imgPath = undefined,
+    light = undefined,
+  }) {
+    if (!imgPath) {
+      imgPath = new Settings().lootIcon;
+    }
+    if (!light) {
+      light = {
+        dim: 0.2,
+        bright: 0.2,
+        luminosity: 0,
+        alpha: 1,
+        color: '#ad8800',
+        coloration: 6,
+        animation: {
+          // type:"sunburst",
+          type: 'radialrainbow',
+          speed: 3,
+          intensity: 10,
+        },
+      };
+    }
+    await ItemPiles.API.turnTokensIntoItemPiles(token);
```

Comment:
I would recommend passing the token update as the secondary argument like:
```js
game.itempiles.API.turnTokensIntoItemPiles(token, {
    tokenSettings: {
        texture: {
            src: imgPath,
        },
        rotation: 0,
    }
});
```

Reply:
See:
https://fantasycomputer.works/FoundryVTT-ItemPiles/#/api?id=turntokensintoitempiles

Reply:
ty for point this out. i will check out.

Reply:
> I would recommend passing the token update as the secondary argument like:
> 
> ```js
> game.itempiles.API.turnTokensIntoItemPiles(token, {
>     tokenSettings: {
>         texture: {
>             src: imgPath,
>         },
>         rotation: 0,
>     }
> });
> ```

sorry @Haxxer i must be very stupid... this piece of code seem not to work:
https://github.com/p4535992/dfreds-pocket-change/blob/f79c52908c4c05352f836ea794b08804d991a2ea/scripts/pocket-change.js#L480

instead of putting the image I indicated it seems to put the image of the first item it extracts from the actor, am I doing something wrong ? 
The light work fine ^^.

Reply:
You want to override the `pileData` as well, as it uses the flags from the default item pile actor, which typically overrides the token image.

Reply:
[This has all of the flags you can set](https://fantasycomputer.works/FoundryVTT-ItemPiles/#/constants?id=pile-flag-defaults), you're looking for `displayOne`, `showItemName`, and `overrideSingleItemScale`, you'll just need to set them to `false`